src/Microsoft.ML.Data/Data/DataViewUtils.cs

```diff
@@ -402,7 +402,7 @@ private static DataViewRowCursor ConsolidateCore(IChannelProvider provider, Data
                                             // The waiter event should never be null since this is only
                                             // called after a point where waiter.Register has been called.
                                             ch.AssertValue(waiterEvent);
-                                        waiterEvent.Wait();
+                                        waiterEvent.Wait(5*60*1000);
                                         waiterEvent = null;
```

Comment:
Similar comment as the one in OrderedWaiter.cs. If the same constant applies everywhere we should use a name.